test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -458,12 +457,15 @@ def closure():
         else:
             optimizer_state_dict = {}
 
-        optimizer_state_dict = _broadcast_object(
-            optimizer_state_dict,
-            src_rank=RECIPIENT_RANK,
-            group=dist.group.WORLD,
-            device=self.device,
-        )
+        optimizer_state_dict_list = [optimizer_state_dict]
+        with torch.cuda.device(self.device):
+            dist.broadcast_object_list(
+                optimizer_state_dict_list,
+                src=RECIPIENT_RANK,
```

Comment:
nit and not related to this PR, but wonder why this name is recipient_rank? This is the rank that's broadcasting its state dict, so all other ranks should technically be the recipient?

Reply:
Yes, this is a bit confusing. `RECIPIENT_RANK` carries its meaning in the lines prior when the local optimizer states are consolidated onto `RECIPIENT_RANK`. Then, this last part of the test checks that loading the consolidated optimizer state on the non-recipient ranks does not throw an exception. The broadcast here is to get the consolidated optimizer state on `RECIPIENT_RANK` to the other ranks, which is why `src=RECIPIENT_RANK`.